### PR TITLE
Add image upload feature with Blobstore

### PR DESCRIPTION
**Pull request by teefang**
Comments can now upload image files

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -45,6 +64,6 @@ window.onload = function() {
 
   var close = document.getElementsByClassName("close")[0];
   close.onclick = function() {
-    modal.style.display= "none";
+  modal.style.display= "none";
```
Comment (0charleschen0) [inline portfolio/src/main/webapp/script.js:67]: Please revert this change.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -31,66 +31,113 @@
...
+    if (value == null) {
+    return defaultValue;
+    }
+    return value;
+  }
 
-      Entity messageEntity = new Entity("comment");
-      messageEntity.setProperty("message", message);
-      messageEntity.setProperty("timestamp", timestamp);
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
```
Comment (0charleschen0) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:111]:
```suggestion
  /** Returns a URL that points to the uploaded file, or {@code null} if the user didn't upload a file. */
```

**Review by 0charleschen0 — APPROVED** [review]
Some nits. Others LGTM.

**Comment by teefang** [comment]
Add upload image feature
